client/service/src/lib.rs

```diff
@@ -295,161 +287,65 @@ async fn build_network_future<
 // Wrapper for HTTP and WS servers that makes sure they are properly shut down.
 mod waiting {
 	pub struct HttpServer(pub Option<sc_rpc_server::HttpServer>);
-	impl Drop for HttpServer {
-		fn drop(&mut self) {
-			if let Some(server) = self.0.take() {
-				server.close_handle().close();
-				server.wait();
-			}
-		}
-	}
 
-	pub struct IpcServer(pub Option<sc_rpc_server::IpcServer>);
-	impl Drop for IpcServer {
+	impl Drop for HttpServer {
 		fn drop(&mut self) {
 			if let Some(server) = self.0.take() {
-				server.close_handle().close();
-				let _ = server.wait();
+				let _ = server.stop().map(|stop| futures::executor::block_on(stop));
```

Comment:
This will spawn a `futures-executor` to complete the task, but doesn't this task need `tokio`?

Reply:
Hmm that looks error prone. Could we not directly call `stop`?